components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilTest.java

```diff
@@ -18,18 +18,206 @@
+
+        mockStatic(IdentityProviderManager.class);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+        when(identityProviderManager.getResidentIdP(anyString())).thenReturn(identityProvider);
+    }
+
+    @Test
+    public void testGetIssuer() throws Exception {
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuer();
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Super tenant mode.");
+        assertEquals(issuer.getFormat(), SAMLSSOConstants.NAME_ID_POLICY_ENTITY,
+                "Issuer format should always be SAML2 spec compatible");
+    }
+
+    @Test
+    public void testGetIssuerFromTenantDomain() throws Exception {
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuerFromTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Issuer for uuper tenant domain.");
+    }
+
+    @Test
+    public void testGetDestinationFromTenantDomain() throws Exception {
+    }
+
+    @Test
+    public void testBuildSAMLAssertion() throws Exception {
+    }
+
+    @Test
+    public void testValidateAuthnRequestSignature() throws Exception {
+    }
+
+    @Test
+    public void testValidateLogoutRequestSignature() throws Exception {
+    }
+
+    @Test
+    public void testValidateDeflateSignature() throws Exception {
+    }
+
+    @Test
+    public void testValidateXMLSignature() throws Exception {
+    }
+
+    @Test
+    public void testBuildErrorResponse() throws Exception {
+    }
+
+    @Test
+    public void testBuildCompressedErrorResponse() throws Exception {
+    }
+
+    @Test
+    public void testValidateACS() throws Exception {
+    }
+
+    @Test
+    public void testUnmarshall() throws Exception {
+
+        XMLObject xmlObject = SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST);
+        assertEquals(xmlObject.getDOM().getAttributeNode("Destination").getValue(),
+                "https://localhost:9443/samlsso");
+        assertEquals(xmlObject.getDOM().getAttributeNode("Reason").getValue(), "Single Logout");
+    }
+
+    @Test
+    public void testMarshall() throws Exception {
+
+        assertEquals(SAMLSSOUtil.marshall(SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST)),
+                TestConstants.DECODED_POST_AUTHN_REQUEST);
+    }
+
+    @Test
+    public void testEncode() throws Exception {
+
+        String encodedXml = SAMLSSOUtil.encode(TestConstants.AUTHN_FAILED_SAML_RESPONSE);
+        assertFalse(encodedXml.isEmpty(), "Encoded xml should not be empty.");
+        assertFalse(encodedXml.contains(" "), "Encoded xml should not contain spaces.");
+    }
+
+    @Test(expectedExceptions = IdentityException.class)
+    public void testDecode() throws Exception {
+
+        assertEquals(SAMLSSOUtil.decode(TestConstants.ENCODED_REDIRECT_AUTHN_REQUEST),
```

Comment:
A description message would be ideal here as well

Reply:
done